sdk/enoki/src/react.tsx

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useStore } from '@nanostores/react';
+import type { ReactNode } from 'react';
+import { createContext, useContext, useState } from 'react';
+
+import type { EnokiFlowConfig } from './EnokiFlow.js';
+import { EnokiFlow } from './EnokiFlow.js';
+
+const EnokiFlowContext = createContext<EnokiFlow | null>(null);
+
+export interface EnokiFlowProviderProps extends EnokiFlowConfig {
+	children: ReactNode;
+}
+
+export function EnokiFlowProvider({ children, ...config }: EnokiFlowProviderProps) {
+	const [enokiFlow] = useState(() => new EnokiFlow(config));
+	return <EnokiFlowContext.Provider value={enokiFlow}>{children}</EnokiFlowContext.Provider>;
+}
+
+export function useEnokiFlow() {
+	const context = useContext(EnokiFlowContext);
+	if (!context) {
+		throw new Error('Missing `EnokiFlowContext` provider');
+	}
+	return context;
+}
+
+export function useZkLogin() {
```

Comment:
I suppose at this point developers still have to be conscious of what zkLogin is and how it works, do we have plans to abstract some of this away from the consumer perspective? I'm not sure what context this will be used in, but if I were a consumer I feel like I wouldn't care what zkLogin is & I'd just want my login state info 😆 

that's my very roundabout way of saying - would it be fair to call this `useLoginInfo`?

Reply:
This is a good callout but I think the reality right now is that we do depend on some level of understanding of what all of these pieces are. I'd argue we're not _really_ a login or auth provider, even though we kind of act and look like one.

I think longer term, this should be our goal, but I think there's a lot of other pieces we need to figure out here to make that happen.